langroid/agent/special/neo4j/neo4j_chat_agent.py

```diff
@@ -31,10 +31,17 @@
 # TOOLS to be used by the agent
 
 
-class CypherQueryTool(ToolMessage):
+class CypherRetrievalTool(ToolMessage):
     request: str = "retrieval_query"
-    purpose: str = """Use this tool to send the Generated Cypher query based on 
-    provided text description and schema."""
+    purpose: str = """Use this tool to send the Cypher query to retreive data from the 
+    graph database based provided text description and schema."""
+    cypher_query: str
+
+
+class CypherCreationTool(ToolMessage):
```

Comment:
@pchalasani noticed you're making changes on the agent, you may want to incorporate this tool. The CypherQueryTool (changed to CypherRetrievalTool in this PR) tool is supposed to be used for retrieval operations, while this one calls `write_query`.